### PR TITLE
[LV]Process alloca in isPredicatedInst for tail-folded analysis.

### PR DESCRIPTION
**Pull request by alexey-bataev**
Patch fixes the compiler crash when it tries to check is alloca in the
loop is a predicated instruction.

### CODE REVIEW

**Comment by alexey-bataev** [comment]
Ping!

File: llvm/test/Transforms/LoopVectorize/tail-folding-alloca-in-loop.ll [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+for.body:
+  %indvars.iv = phi i64 [ 0, %entry ], [ %indvars.iv.next, %for.body ]
+  %0 = alloca i8, i64 %n, align 16
+  %arrayidx = getelementptr inbounds ptr, ptr %vf1, i64 %indvars.iv
+  store ptr %0, ptr %arrayidx, align 8
+  %indvars.iv.next = add i64 %indvars.iv, 1
+  %exitcond.not = icmp eq i64 %indvars.iv, 200
+  br i1 %exitcond.not, label %for.end.loopexit, label %for.body
+
+for.end.loopexit:
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/tail-folding-alloca-in-loop.ll:83]: Nit: can be shortened to exit

**Review by fhahn — APPROVED** [review]
LGTM, thanks should definitely not crash. Would
Be interesting to know if there are any real world cases where alloca are in the loop body to double check we are doing the right thing in general

**Comment by alexey-bataev** [comment]
> LGTM, thanks should definitely not crash. Would Be interesting to know if there are any real world cases where alloca are in the loop body to double check we are doing the right thing in general

The test itself is a reduced version from llvm-test-suite, compiled with folded tail.

**Pull request merged**